usr/sbin/p11sak/p11sak.c

```diff
@@ -1819,11 +1856,18 @@ static CK_RV tok_key_get_key_type(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE hk
  * Check args for gen_key command.
  */
 static CK_RV check_args_gen_key(p11sak_kt *kt, CK_ULONG keylength,
-                                char *ECcurve)
+                                char *ECcurve, char *dilithium_ver)
 {
     switch (*kt) {
     case kt_DES:
     case kt_3DES:
+    case kt_IBM_DILITHIUM:
+        if (strcmp(dilithium_ver, "r2_65") == 0) {
```

Comment:
strcasecmp here as well ? I would treat the Dilithium version as non-case-sensitive.

Reply:
done